lib/oli_web/live/delivery/remix_section.ex

```diff
@@ -848,6 +848,49 @@ defmodule OliWeb.Delivery.RemixSection do
     {:noreply, hide_modal(socket, modal_assigns: nil)}
   end
 
+  def handle_event("show_hide_resource_modal", %{"uuid" => uuid}, socket) do
+    %{hierarchy: hierarchy} = socket.assigns
+
+    node = Hierarchy.find_in_hierarchy(hierarchy, uuid)
+
+    modal_assigns = %{
+      id: "hide_#{uuid}",
+      node: node
+    }
+
+    modal = fn assigns ->
+      ~H"""
+      <HideResourceModal.render {@modal_assigns} />
+      """
+    end
+
+    {:noreply,
+     show_modal(
+       socket,
+       modal,
+       modal_assigns: modal_assigns
+     )}
+  end
+
+  def handle_event("HideResourceModal.toggle", %{"uuid" => uuid}, socket) do
+    %{hierarchy: hierarchy, section: section} = socket.assigns
+
+    section_resource = Hierarchy.find_in_hierarchy(hierarchy, uuid).section_resource
+
+    Sections.update_section_resource(section_resource, %{hidden: !section_resource.hidden})
```

Comment:
The toggling of `hidden` cannot be directly made to the section resource record.  Take a look at how reordering and how removing (aka "deleting") resources is done.  The change is made "in memory" to the `hierarchy` assign, and the `has_unsaved_changes` flag is then set to `true`.   Toggling `hidden` should fit into this same framework.